homeassistant/components/weheat/sensor.py

```diff
@@ -0,0 +1,65 @@
+"""Platform for sensor integration."""
+
+from decimal import Decimal
+
+from weheat.abstractions.heat_pump import HeatPump
+
+from homeassistant.components.sensor import SensorEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN, SENSORS
+from .coordinator import WeheatDataUpdateCoordinator
+from .entity import WeheatEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the sensors for weheat heat pump."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities(
+        WeheatHeatPumpSensor(
+            coordinator=coordinator, entity_description=entity_description
+        )
+        for entity_description in SENSORS
+        if hasattr(HeatPump, entity_description.key)
+    )
+
+
+class WeheatHeatPumpSensor(WeheatEntity, SensorEntity):
+    """An entity using CoordinatorEntity.
+
+    The CoordinatorEntity class provides:
+      should_poll
+      async_update
+      async_added_to_hass
+      available
+
+    """
```

Comment:
Not really needed imo, because this can change in the future and it would run stale